src/com/ms/silverking/cloud/storagepolicy/StoragePolicyZK.java

```diff
@@ -16,44 +16,44 @@
 
 public class StoragePolicyZK extends MetaToolModuleBase<StoragePolicyGroup, MetaPaths> {
 
-    public StoragePolicyZK(MetaClientBase<MetaPaths> mc) throws KeeperException {
-        super(mc, mc.getMetaPaths().getStoragePolicyGroupPath());
+  public StoragePolicyZK(MetaClientBase<MetaPaths> mc) throws KeeperException {
+    super(mc, mc.getMetaPaths().getStoragePolicyGroupPath());
+  }
+
+  @Override
+  public StoragePolicyGroup readFromFile(File file, long version) throws IOException {
+    return new PolicyParser().parsePolicyGroup(file, version);
+  }
+
+  @Override
+  public StoragePolicyGroup readFromZK(long version, MetaToolOptions options) throws KeeperException {
+    try {
+      String vPath;
+
+      vPath = getVersionPath(version);
+      return new PolicyParser().parsePolicyGroup(zk.getString(vPath), version);
+    } catch (PolicyParseException ppe) {
+      throw new RuntimeException(ppe);
     }
-
-    @Override
-    public StoragePolicyGroup readFromFile(File file, long version) throws IOException {
-        return new PolicyParser().parsePolicyGroup(file, version);
-    }
-
-    @Override
-    public StoragePolicyGroup readFromZK(long version, MetaToolOptions options) throws KeeperException {
-        try {
-            String          vPath;
-            
-            vPath = getVersionPath(version);
-            return new PolicyParser().parsePolicyGroup(zk.getString(vPath), version);
-        } catch (PolicyParseException ppe) {
-            throw new RuntimeException(ppe);
-        }
-    }
-
-    @Override
-    public void writeToFile(File file, StoragePolicyGroup policyGroup) throws IOException {
-        BufferedWriter  writer;
-        
-        writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(file)));
-        writer.write(policyGroup.getPolicy(policyGroup.getName()).toString());
-        for (StoragePolicy policy : policyGroup.getPolicies()) {
-            if (!policy.getName().equals(policyGroup.getName())) {
-                writer.write(policy.toString());
-            }
-        }
-        writer.close();
-    }
-
-    @Override
-    public String writeToZK(StoragePolicyGroup policyGroup, MetaToolOptions options) throws IOException, KeeperException {
-        zk.createString(base +"/" , policyGroup.toString(), CreateMode.PERSISTENT_SEQUENTIAL);
-        return null;
+  }
+
+  @Override
+  public void writeToFile(File file, StoragePolicyGroup policyGroup) throws IOException {
+    BufferedWriter writer;
+
+    writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(file)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *writer*. The resource is referenced in statements at the following lines: 44, 45, 48. The resource closure statement is at line: 51. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Either a) close *writer* in a try-finally block or b) close the resource by declaring *writer* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).